COMP353-Project/CON
# Set up authentication with jwt, signin, signout and a private/public navbar

## Contents of this PR:

- Implemented login using context.
- Refactored the NavBar so that it displays different elements depending on whether the user is logged in/ a guest.
- Added logout button that is fully functional.

### How to test:

1. Create a user in the database (the password needs to be hashed).
2. Start the app.
3. Click on the Login button (top right).
4. Enter your email + password.
5. You should be redirected to the Condo Association page & the navbar should have more items on it.
6. To logout, click on the logout button (top right).
7. You should be redirected to the public homepage.

### Note: 
I pulled from @kamilgeagea's state management branch to be able to use the Authentication Context.
